web-client/src/modules/create/components/NewAddressModal.tsx

```diff
@@ -0,0 +1,158 @@
+import { Checkbox, Col, Form, Input, Modal, Row } from 'antd';
+import { useForm } from 'antd/lib/form/Form';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { GeocoderComponent } from 'src/components/Geocoder/Geocoder';
+import { updateUserPrivilegedInformation } from 'src/ducks/profile/actions';
+import { IUserAddress } from 'src/models/users/privilegedInformation';
+import { AppState } from 'src/store';
+
+const ModalForm: React.FC<ModalFormProps> = ({
+  geocode,
+  visible,
+  closeModal,
+  modalSuccess,
+}) => {
+  const [form] = useForm();
+  const dispatch = useDispatch();
+
+  const [shouldSave, setShouldSave] = useState<boolean>(false);
+
+  const addresses = useSelector(
+    (state: AppState) => state.profile.privilegedInformation?.addresses,
+  );
+  const privilegedInformation = useSelector(
+    (state: AppState) => state.profile.privilegedInformation,
+  );
+  const userId = useSelector((state: AppState) => state.auth.user?.uid);
+
+  const handleFinish = () => {
```

Comment:
Components generally should not interact with redux or call services like geocode directly--that is the job of the container. This separation makes it possible in principle to render components in isolation, without needing to mock the store or set up application state. Consider passing the address object in modalSuccess, and bubbling it back to the container, where it calls geocode and dispatches the update

Reply:
This wrapped component interacts directly with the geocoder.   The CreatePostContainer does not need to know about it.